### PR TITLE
Ancient lens and Truestrike lens rework

### PR DESCRIPTION
**Pull request by ThatDamnPineapple**
No trello card, but reworks ancient lens and truestrike lens

Nothing more to do as far as I know

### CODE REVIEW

File: Content/Items/UndergroundTemple/TempleLensSystem.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+		private void Main_CheckMonoliths(On.Terraria.Main.orig_CheckMonoliths orig)
+		{
+			GraphicsDevice gD = Main.graphics.GraphicsDevice;
+			SpriteBatch spriteBatch = Main.spriteBatch;
+
+			if (Main.gameMenu || Main.dedServ || spriteBatch is null || NPCTarget is null || gD is null || (!Main.npc.Any(n => n.active && n.HasBuff(ModContent.BuffType<Buffs.Illuminant>()))) && !Main.npc.Any(n => n.active && n.HasBuff(ModContent.BuffType<Exposed>())))
+				return;
+
+			#region npctarget
```
Comment (ScalarVector1) [inline Content/Items/UndergroundTemple/TempleLensSystem.cs:46]: Is there a reason for a region thats a grand total of 4 lines shorter than the entire method? Seems a bit redundant and pointlessly confusing

File: Content/Items/UndergroundTemple/Accessories.TempleLensUpgrade.cs [hunk 2]
```diff
@@ -31,13 +33,20 @@ public override void Load()
 		private void ModifyHurtLens(Player Player, Item Item, NPC target, ref int damage, ref float knockback, ref bool crit)
 		{
 			if (Equipped(Player) && crit)
-				target.AddBuff(BuffType<Exposed>(), 120);
+				ApplyExposed(target);
```
Comment (ScalarVector1) [inline Content/Items/UndergroundTemple/Accessories.TempleLensUpgrade.cs:36]: Should we consider switching over to the exposure system added by Skippz a while back? That way we can take advantage of proper stacking and keywords.

File: Content/Items/UndergroundTemple/TempleLensSystem.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+			Main.spriteBatch.Begin(default, BlendState.AlphaBlend, default, default, default, null, Main.GameViewMatrix.ZoomMatrix);
+
+			Texture2D bloom = ModContent.Request<Texture2D>(AssetDirectory.Keys + "GlowAlpha").Value;
+			Color gold = Color.Orange;
+			gold.A = 0;
+			for (int i = 0; i < Main.npc.Length; i++)
+			{
+				NPC NPC = Main.npc[i];
+
+				if (NPC.active && NPC.HasBuff(ModContent.BuffType<Buffs.Illuminant>()))
```
Comment (ScalarVector1) [inline Content/Items/UndergroundTemple/TempleLensSystem.cs:90]: single line conditionals should be inlined

File: Content/Items/UndergroundTemple/TempleLensSystem.cs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+			}
+			spriteBatch.End();
+			gD.SetRenderTargets(bindings);
+			#endregion
+		}
+
+		private void DrawLens(On.Terraria.Main.orig_DrawNPCs orig, Main self, bool behindTiles)
+		{
+			orig(self, behindTiles);
+			if (!behindTiles && (Main.npc.Any(n => n.active && n.HasBuff(ModContent.BuffType<Buffs.Illuminant>())) || Main.npc.Any(n => n.active && n.HasBuff(ModContent.BuffType<Exposed>()))))
```
Comment (ScalarVector1) [inline Content/Items/UndergroundTemple/TempleLensSystem.cs:108]: Should we consider making a static property for this so we dont have to copy/paste this complex condition around?

File: Content/Items/UndergroundTemple/TempleLensSystem.cs (hunk 1 above)
Comment (ThatDamnPineapple) [reply Content/Items/UndergroundTemple/TempleLensSystem.cs:50]: Fixed

File: Content/Items/UndergroundTemple/TempleLensSystem.cs (hunk 3 above)
Comment (ThatDamnPineapple) [reply Content/Items/UndergroundTemple/TempleLensSystem.cs:98]: Fixed

File: Content/Items/UndergroundTemple/Accessories.TempleLensUpgrade.cs (hunk 2 above)
Comment (ThatDamnPineapple) [reply Content/Items/UndergroundTemple/Accessories.TempleLensUpgrade.cs:36]: Seems like more of a design discussion. Ping me on the dev discord when you read this and we can discuss with egshels

File: Content/Items/UndergroundTemple/TempleLensSystem.cs (hunk 1 above)
Comment (ScalarVector1) [reply Content/Items/UndergroundTemple/TempleLensSystem.cs:51]: Lets replace these regions with comments -- they're entirely contained within one method making the use case of collapsing them not very useful

File: Effects/Compile/TempleLens.fx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+texture LensTarget;
+sampler lens = sampler_state
+{
+    Texture = (LensTarget);
+};
+
+float4 White(float4 sampleColor : COLOR0, float2 coords : TEXCOORD0) : COLOR0
+{
+	float4 color2 = tex2D(lens, coords);
```
Comment (ScalarVector1) [inline Effects/Compile/TempleLens.fx:24]: seems like there is an extra tab here

File: Core/IResizable.cs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿namespace StarlightRiver.Core
+{
+	interface IResizable
+	{
+		public void ResizeTarget() { }
+
+		public bool IsResizable { get; }
```
Comment (ScalarVector1) [inline Core/IResizable.cs:7]: Why would something ever be an IResizable but not be resizable? Is this more intended as a 'should resize' switch? If it is this name seems a bit in conflict with the name of the interface

File: StarlightRiver.cs [hunk 7]
```diff
@@ -28,6 +29,8 @@ public partial class StarlightRiver : Mod
 	{
 		private List<IOrderedLoadable> loadCache;
 
+		private List<IResizable> resizableCache;
```
Comment (ScalarVector1) [inline StarlightRiver.cs:32]: This should be its own seperate ModSystem, not part of the main mod class.

**Review by ScalarVector1 — CHANGES_REQUESTED** [review]

**Review by ScalarVector1 — COMMENTED** [review]
This looks mostly fine from looking it over, but I did alot of manual porting work over to the new ScreenTarget system. Another look over this is warranted for sure.

**Review by ScalarVector1 — APPROVED** [review]

**Pull request merged**